spec/support/wait_for_ajax.rb

```diff
@@ -1,11 +1,18 @@
 module WaitForAjax
   def wait_for_ajax
-    Timeout.timeout(Capybara.default_max_wait_time) do
-      loop until finished_all_ajax_requests?
+    max_time = Capybara::Helpers.monotonic_time + Capybara.default_max_wait_time
+    while Capybara::Helpers.monotonic_time < max_time
+      finished = finished_all_ajax_requests?
+      if finished
+        break
+      else
+        sleep 0.1
+      end
     end
+    raise 'wait_for_ajax timeout' unless finished
```

Comment:
oooh, interesting. i did a quick googling and didn't find anything. but i will def keep that in mind for the frontend redo since all the frontend tests are gonna need to be addressed.